src/App.tsx

```diff
@@ -1,26 +1,125 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import { deleteTodo, getTodos, createTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { TodoCompletedCategory } from './types/TodoCompletedCategory';
+import { filterTodosByComplated } from './utils/filterTodosByCompleted';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [completedCategory, setCompletedCategory] =
+    useState<TodoCompletedCategory>(TodoCompletedCategory.all);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.noneError);
+  const [removedTodosId, setRemovedTodosId] = useState([0]);
+
+  const filtredTodos = filterTodosByComplated(todos, completedCategory);
+  const countOfCompletedTodos = todos.filter(todo => todo.completed).length;
+  const countOfNotCompletedTodos = todos.length - countOfCompletedTodos;
+  const isSomeTodoComplated = todos.some(todo => todo.completed);
+
+  async function fetchTodosFromApi() {
+    try {
+      const todosFromApi = await getTodos();
+
+      setTodos(todosFromApi);
+    } catch {
+      setErrorMessage(Errors.loadError);
+    }
+  }
+
+  async function fetchAddTodo(newTitle: string) {
+    let newTodo: Todo = {
+      id: 0,
+      title: newTitle,
+      userId: 0,
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    try {
+      newTodo = await createTodo(newTitle);
+    } catch {
+      setErrorMessage(Errors.addError);
+    } finally {
+      setTempTodo(null);
+      if (newTodo.id !== 0) {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+
+        return true;
+      } else {
+        return false;
+      }
+    }
```

Comment:
The loader doesn't work during adding a new todo.
You can add the loader using `removedTodosId` (maybe it would be better to rename it). Just add new id to this array before sending a request and delete this id after, as well as in `fetchDeleteTodo`.